crates/wasmtime/src/runtime/component/func/typed.rs

```diff
@@ -914,6 +915,44 @@ macro_rules! floats {
                 *ptr = self.to_bits().to_le_bytes();
                 Ok(())
             }
+
+            fn store_list<T>(
+                cx: &mut LowerContext<'_, T>,
+                ty: InterfaceType,
+                offset: usize,
+                items: &[Self],
+            ) -> Result<()> {
+                debug_assert!(matches!(ty, InterfaceType::$ty));
+
+                // Double-check that the CM alignment is at least the host's
+                // alignment for this type which should be true for all
+                // platforms.
+                assert!((Self::ALIGN32 as usize) >= mem::align_of::<Self>());
+
+                // Slice `cx`'s memory to the window that we'll be modifying.
+                // This should all have already been verified in terms of
+                // alignment and sizing meaning that these assertions here are
+                // not truly necessary but are instead double-checks.
+                //
+                // Note that we're casting a `[u8]` slice to `[$integer]` (with
+                // $integer having the same size as Self) with `align_to_mut`
+                // which is not safe in general but is safe in our specific
+                // case as all `u8` patterns are valid `$integer` patterns
+                // since `$integer` is an integral type.
+                let dst = &mut cx.as_slice_mut()[offset..][..items.len() * Self::SIZE32];
+                let (before, middle, end) = unsafe { dst.align_to_mut::<$integer>() };
```

Comment:
```suggestion
                let (before, middle, end) = unsafe { dst.align_to_mut::<$float>() };
```

Presumably the alignment of e.g. `f32` and `u32` should be the same, but we might as well be precise here (and in similar code below).

Reply:
The reason I used integer types and `align_to[_mut]::<$integer>` was to handle endianness correctly.
I felt like using `align_to[_mut]::<$float>` and thereby transmuting the component's memory to `$float` is incorrect, because (in my understanding) `$float` represents floating point numbers in native endianness, while `dst` is a slice of floating point numbers with little-endian byte-order. 

Therefore, I thought using `$float` for floating point numbers with different endianess would also cause confusion, in particular when converting endianess. 

I also considered using only accordingly aligned bytes (e.g. `[u8; 4]`) for the memory of the component, but I was unsure if that would enable the same optimizations. 
Is there any easy (built-in) way to benchmark Wasmtime in specific scenarios?

Let me know what you think.